### ISSUE TITLE
SKR1.4-Turbo + TMC2209 + Marlin 2.0.x + StealthChop issue

### ISSUE THREAD

**Issue by DustinSnijders** [bigtreetech/BIGTREETECH-SKR-V1.3#357]
Hi All,

I'm running a configuration as above (skr1.4 + tmc2209) on Marlin 2. In advance everything seems to work fine, no compiling issues and all the M-commands works as expected. In past I used a SKR1.3 board with TMC2130's. No issue at all. I only switched the board and the steppendrivers right now, due to some heath issues (Delta printer) with the TMC2130's and I like te reuse the board for another (carthesian) printer.

What is the issue; when I print with SpreadCycle activated, everything is okay. Standard deviation is very ok (0.006) and the printresults are like I expect...even better. The SpreadCycle is a bit noisy, you can almost hear the data whispering, but still better than without compensation... :-) When I activate StealthChop (M569 S1 X Y Z) an oase of silence, but if I send for example G28 or a G0 Z50 to the  printer, it moves nice and without noise, in a completely weird direction, unlogical sequenced, unpredictable and finishes with an error: printer halted.

Does anyone recognise this issue? SpreadCycle: Working top (bit noisy) and SteathChop: quiet but not manageble en giving errors like: 

Recv: Error:Printer halted. kill() called!
Changing monitoring state from "Operational" to "Error: Printer halted. kill() called!"
Send: M112
Send: N2 M112*35
Send: N3 M104 T0 S0*34
Send: N4 M140 S0*97
Changing monitoring state from "Error: Printer halted. kill() called!" to "Offline (Error: Printer halted. kill() called!)"

I attached my conf files:

[Marlin 2.0.6.zip](https://github.com/bigtreetech/BIGTREETECH-SKR-V1.3/files/4804341/Marlin.2.0.6.zip)

I also tried the not-nightly and not-bèta versions of Marlin, but there was no difference. I hope that someone can help me.

Cheers,
Dustin

**Comment by DustinSnijders** [comment]
By the way, I don't see errors when I switch to StealthChop. Really weird. Someone any clue?

**Comment by DustinSnijders** [comment]
Update: I changed the TCM2209's for my older TCM2130's...exactly te same issue as described above...
SPI instead of UART, no difference and still no clue...help?!

**Comment by DustinSnijders** [comment]
Update: Reflowing the LPC1769 did not help as well...

I had a brainwave; UART control of the stepperdrivers is using a serial bus. My display is using a serial connection as well, maybee is both  the data conflicting...nope. I even did 'greyed out'  the display, but no solution...

**Comment by DustinSnijders** [comment]
Many hours later....I was apparently searching in the wrong direction. At the end I decided to use another earlier distri I used before with my SKR1.3.  It is a combination of the Bigtreetech distri and Marlin 2.0.5 examples config files for my Delta (Kossel XL 80cm) I merged in the past. I changed the board and some other small settings in relation to the LPC1769 and the 1.4 HW
I'm also using the rest of the source of the 2.0.5 Marlin distri. It seems that there is a slight difference what haves effect on the stealthchop function. It works fine now...
